### ISSUE TITLE
Test keep Failing

### ISSUE THREAD

**Issue by rusrushal13** [zulip/python-zulip-api#310]
I wanted to contribute to the repo so I started setting up the environment and every time I run tests, one test kept failing. 
Logtrace: 
```
(zulip-api-py3-venv) servifys-MacBook-Air-4:python-zulip-api servify$ ./tools/test-bots

...
...

test_bot (zulip_bots.bots.weather.test_weather.TestWeatherBot) ... ok
test_bot_responds_to_empty_message (zulip_bots.bots.weather.test_weather.TestWeatherBot) ... ok
test_bot_usage (zulip_bots.bots.weather.test_weather.TestWeatherBot) ... ok
test_bot (zulip_bots.bots.wikipedia.test_wikipedia.TestWikipediaBot) ... ERROR:root:Page not found.
ERROR:root:broken link
ok
test_bot_responds_to_empty_message (zulip_bots.bots.wikipedia.test_wikipedia.TestWikipediaBot) ... ok

...
...
...

test_bot (zulip_bots.bots.yoda.test_yoda.TestYodaBot) ... ERROR:root:Invalid Api Key. Did you follow the instructions in the `readme.md` file?
ERROR:root:Unknown Error.
ok
test_bot_responds_to_empty_message (zulip_bots.bots.yoda.test_yoda.TestYodaBot) ... ok
test_bot_usage (zulip_bots.bots.yoda.test_yoda.TestYodaBot) ... ok
test_bot_responds_to_empty_message (zulip_bots.bots.youtube.test_youtube.TestYoutubeBot) ... ok
test_bot_usage (zulip_bots.bots.youtube.test_youtube.TestYoutubeBot) ... ok
test_connection_error (zulip_bots.bots.youtube.test_youtube.TestYoutubeBot) ... WARNING:root:Bad connection
ok
test_help (zulip_bots.bots.youtube.test_youtube.TestYoutubeBot) ... ok
test_invalid_key (zulip_bots.bots.youtube.test_youtube.TestYoutubeBot) ... ok
test_multiple (zulip_bots.bots.youtube.test_youtube.TestYoutubeBot) ... ok
test_noresult (zulip_bots.bots.youtube.test_youtube.TestYoutubeBot) ... ok
test_single (zulip_bots.bots.youtube.test_youtube.TestYoutubeBot) ... ok
test_extract_query_without_mention (zulip_bots.test_run.TestBotLib) ... ok
test_argument_parsing_with_bot_name (zulip_bots.test_run.TestDefaultArguments) ... ok
test_argument_parsing_with_bot_path (zulip_bots.test_run.TestDefaultArguments) ... ok

======================================================================
FAIL: test_exception_when_api_key_is_invalid (zulip_bots.bots.baremetrics.test_baremetrics.TestBaremetricsBot)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/servify/Documents/python-zulip-api/zulip_bots/zulip_bots/bots/baremetrics/test_baremetrics.py", line 82, in test_exception_when_api_key_is_invalid
    bot_test_instance.initialize(StubBotHandler())
AssertionError: BotQuitException not raised

----------------------------------------------------------------------
Ran 205 tests in 3.760s

FAILED (failures=1)
``` 

Can somebody can explain to me what is happening here?

**Comment by roberthoenig** [comment]
@rusrushal13 thanks for the report! I'm getting the same error in my dev env. Looking into it..

**Comment by roberthoenig** [comment]
Fixed by #308. @rusrushal13 can you update your repo and run the tests again?

**Comment by rusrushal13** [comment]
Cool, now it's running fine. Thanks 😃
